v2/cmd/asoctl/cmd/export_template.go

```diff
@@ -122,3 +137,45 @@ asoctl export template --version v2.6.0 --crd-pattern "resources.azure.com/*;con
 
 	return cmd
 }
+
+func validateCRDPatternsForTemplate(
+	crdPatterns []string,
+	log logr.Logger,
+) {
+	scheme := api.CreateScheme()
+	knownCRDs := scheme.AllKnownTypes()
+	warnings := 0
+	for _, p := range crdPatterns {
+		// We split patterns by `;` so we can check them individually and provide more fine-grained feedback
+		for _, pattern := range strings.Split(p, ";") {
+			count := countCRDsMatchingPattern(knownCRDs, pattern)
+			if count == 0 {
+				log.V(Status).Info("No CRDs matched pattern", "pattern", pattern)
+				warnings++
+			} else {
+				log.V(Info).Info("Pattern matched CRDs", "pattern", pattern, "count", count)
+			}
+		}
+	}
+
+	if warnings > 0 {
+		log.V(Status).Info("Possibly ineffective CRD patterns detected")
```

Comment:
Same comment about us needing to be careful due to piping. Fine if this goes to stderr, let's just confirm that's the case?

Reply:
All logs go to stderr.